core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -19,43 +20,130 @@ import com.unciv.ui.victoryscreen.RankingType
 
 object Automation {
 
-    fun rankTileForCityWork(tile: TileInfo, city: CityInfo, foodWeight: Float = 1f): Float {
+    fun rankTileForCityWork(tile: TileInfo, city: CityInfo): Float {
         val stats = tile.getTileStats(city, city.civInfo)
-        return rankStatsForCityWork(stats, city, foodWeight)
+        return rankStatsForCityWork(stats, city)
     }
 
-    private fun rankStatsForCityWork(stats: Stats, city: CityInfo, foodWeight: Float = 1f): Float {
-        var rank = 0f
+    private fun rankStatsForCityWork(stats: Stats, city: CityInfo): Float {
+        val cityAIFocus = city.cityAIFocus
+
+        val yieldStats = stats.clone()
+        //var happinessYield = stats.happiness  // TODO: count happiness?
+        //if (city.civInfo.getHappiness() < 0) happinessYield *= 2  // double weight for unhappy civilization
+        // var GPPYield = stats.GPP * 3 // TODO: rework GPP Yield
+
+        val surplusFood = city.foodForNextTurn() // intensive to calc based on Excess Food?
+        // Apply base weights
+        yieldStats.applyRankingWeights()
+        
+        if (cityAIFocus != CityFocus.NoFocus && cityAIFocus != CityFocus.FoodFocus && cityAIFocus != CityFocus.ProductionGrowthFocus && cityAIFocus != CityFocus.GoldGrowthFocus){
+            // Focus on non-food/growth
+            if (surplusFood < 0){
+                // Starving, need Food, get to 0
+                yieldStats.food *= 8
+            } else {
+                yieldStats.food /= 2
+            }
+        } else {
+            // NoFocus or Food/Growth Focus. Target +2 Food Surplus
+            if (surplusFood < 2) {
+                yieldStats.food * 8
+            } else if (cityAIFocus != CityFocus.FoodFocus) {
+                yieldStats.food /= 2
+            }
```

Comment:
There's too much indentation here for me, please remove the brackets on the inner ifelses